### ISSUE TITLE
Inefficient buffer growth algorithm

### ISSUE THREAD

**Issue by ghost** [peterjc/backports.lzma#47]
Please read https://bugs.python.org/issue41265

`backports.lzma` has the same problem, [the code](https://github.com/peterjc/backports.lzma/blob/backports.lzma.v0.0.14/backports/lzma/_lzmamodule.c#L216-L227):
```c
    #if BUFSIZ < 8192
    #define INITIAL_BUFFER_SIZE 8192
    #else
    #define INITIAL_BUFFER_SIZE BUFSIZ
    #endif

    static int
    grow_buffer(PyObject **buf)
    {
        size_t size = PyBytes_GET_SIZE(*buf);
        return _PyBytes_Resize(buf, size + (size >> 3) + 6);
    }
```

`backports.lzma` can use a new growth algorithm, and switch to CPython's algorithm after it changes.

**Comment by peterjc** [comment]
If and when CPython's version changes, I am happy to have the changes applied to the backport.

**Comment by peterjc** [comment]
This was resolved upstream in Python 3.10, see https://bugs.python.org/issue41486

I'm about to archive this repository, and so am closing the outstanding issues and pull requests. Thank you for your efforts though.
